integration/commands_administration_test.go

```diff
@@ -773,6 +775,227 @@ func TestCommandsAdministrationDBStatsEmptyWithScale(t *testing.T) {
 	// https://github.com/FerretDB/FerretDB/issues/727
 }
 
+func TestCommandsAdministrationRenameCollection(t *testing.T) {
+	t.Parallel()
+
+	maxTableLen := strings.Repeat("a", 63) // maxTableLen = 63
+	insertCollections := []string{"foo", "buz"}
+
+	for name, tc := range map[string]struct { //nolint:vet // for readability
+		collection       string
+		targetCollection string
+		expected         bson.D
+		err              *mongo.CommandError
+		toInsert         []interface{}
+	}{
+		"Rename": {
+			collection:       "foo",
+			targetCollection: "bar",
+			expected:         bson.D{{"ok", float64(1)}},
+		},
+		"RenameSame": {
+			collection:       "foo",
+			targetCollection: "foo",
+			err: &mongo.CommandError{
+				Code:    20,
+				Name:    "IllegalOperation",
+				Message: `Can't rename a collection to itself`,
+			},
+		},
+		"RenameDuplicate": {
+			collection:       "foo",
+			targetCollection: "buz",
+			err: &mongo.CommandError{
+				Code:    48,
+				Name:    "NamespaceExists",
+				Message: `target namespace exists`,
+			},
+		},
+		"SoureDoesNotExist": {
+			collection:       "none",
+			targetCollection: "bar",
+			err: &mongo.CommandError{
+				Code:    26,
+				Name:    "NamespaceNotFound",
+				Message: "source namespace does not exist",
+			},
+		},
+		// this confirms that after we rename foo to bar and then create foo again,
+		// 1. bool-false doesn't exist
+		// 2. the newly inserted documents exist
+		"InsertIntoOld": {
+			collection:       "foo",
+			targetCollection: "bar",
+			expected:         bson.D{{"ok", float64(1)}},
+			toInsert: []interface{}{
+				bson.D{{"_id", 1}},
+				bson.D{{"_id", 2}},
+			},
+		},
+		"RenameMaxTableLen": {
+			collection:       "foo",
+			targetCollection: maxTableLen + "a",
+			err: &mongo.CommandError{
+				Code:    73,
+				Name:    "InvalidNamespace",
+				Message: "collection is too long",
+			},
+		},
+	} {
+		name, tc := name, tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			ctx, collection := setup.Setup(t, shareddata.Bools)
+
+			db := collection.Database()
+
+			require.NotEmpty(t, insertCollections)
+			for _, coll := range insertCollections {
+				require.NoError(t, db.CreateCollection(ctx, coll))
```

Comment:
I don't understand why we create insertCollections, don't we just need to insert source collection?
`require.NoError(t, db.CreateCollection(ctx, collection))` I prefer naming of `sourceCollection` though :hugs:

Reply:
Ah yes, that is to test we cannot rename if target collection already exists. I forgot to add a test for that hence the confusion. My bad!

Fixed now.